python/cuvs/cuvs/neighbors/brute_force/brute_force.pyx

```diff
@@ -0,0 +1,216 @@
+#
+# Copyright (c) 2024, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# cython: language_level=3
+
+import numpy as np
+
+cimport cuvs.common.cydlpack
+
+from cuvs.common.resources import auto_sync_resources
+
+from cython.operator cimport dereference as deref
+from libc.stdint cimport uint32_t
+from libcpp cimport bool
+
+from cuvs.common cimport cydlpack
+from cuvs.distance_type cimport DistanceType
+
+from pylibraft.common import auto_convert_output, cai_wrapper, device_ndarray
+from pylibraft.common.cai_wrapper import wrap_array
+from pylibraft.common.interruptible import cuda_interruptible
+from pylibraft.distance.pairwise_distance import DISTANCE_TYPES
+from pylibraft.neighbors.common import _check_input_array
+
+from cuvs.common.c_api cimport cuvsResources_t
+
+from cuvs.common.exceptions import check_cuvs
+
+
+cdef class FlatIndex:
+    """
```

Comment:
I understand the reasoning behind changing this to flat index, but I don't think this is the right way to do this and I'm not sure if we need to follow FAISS' naming convention here. We call this index brute-force everywhere else, so we should stick that for consistency.